### PR TITLE
AG-7933 - Bugfixes for BarSeries/AreaSeries simplifications.

### PR DESCRIPTION
**Pull request by alantreadway**
https://ag-grid.atlassian.net/browse/AG-7933

Multiple fixes:
- Fixes cross-filtering bar/column series rendering so that filtered-out bar appears correctly.
- Fixes cross-filtering and animation updates not triggering redraw.
- Fixes incorrect grouping of ungrouped series types, leading to weird rendering.

Bouns:
- Adds regression tests for integrated-charts cases that triggered some of these cases.

### CODE REVIEW

File: charts-community-modules/ag-charts-community/src/scene/group.ts [hunk 1]
```diff
@@ -271,7 +271,7 @@ export class Group extends Node {
         if (hasVirtualChildren) {
             // Mark virtual nodes as clean, but don't recurse (their children have already been updated).
             for (const child of this.virtualChildren) {
-                child.markClean({ recursive: false });
+                child.markClean({ recursive: 'virtual' });
```
Comment (lsjroberts) [inline charts-community-modules/ag-charts-community/src/scene/group.ts:274]: nit: The comment is not entirely accurate I guess? `..., and only recurse their virtual children ...` or something?

File: charts-community-modules/ag-charts-community/src/chart/__image_snapshots__/integrated-charts-examples-test-ts-integrated-charts-examples-changing-chart-type-for-histogram-it-should-render-to-canvas-as-expected-1-snap.png
Comment (lsjroberts) [inline charts-community-modules/ag-charts-community/src/chart/__image_snapshots__/integrated-charts-examples-test-ts-integrated-charts-examples-changing-chart-type-for-histogram-it-should-render-to-canvas-as-expected-1-snap.png:1]: Is this as expected? Not too familiar with histograms but looks a bit off?

**Review by lsjroberts — APPROVED** [review]
LGTM, great stuff

File: charts-community-modules/ag-charts-community/src/chart/__image_snapshots__/integrated-charts-examples-test-ts-integrated-charts-examples-changing-chart-type-for-histogram-it-should-render-to-canvas-as-expected-1-snap.png
Comment (alantreadway) [reply charts-community-modules/ag-charts-community/src/chart/__image_snapshots__/integrated-charts-examples-test-ts-integrated-charts-examples-changing-chart-type-for-histogram-it-should-render-to-canvas-as-expected-1-snap.png:1]: Yeah I also paused on this one - I've just verified on production and the same data-set renders the same, so I'm ruling it out of scope to fix as part of this PR.

**Pull request merged**